cmd/ctr/commands/tasks/kill.go

```diff
@@ -74,10 +79,34 @@ var killCommand = cli.Command{
 		if err != nil {
 			return err
 		}
+		if context.String("signal") != "" {
+			signal, err = commands.ParseSignal(context.String("signal"))
+			if err != nil {
+				return err
+			}
+		} else {
+			signal, err = getStopSignal(ctx, container, signal)
+			if err != nil {
+				return err
+			}
+		}
 		task, err := container.Task(ctx, nil)
 		if err != nil {
 			return err
 		}
 		return task.Kill(ctx, signal, opts...)
 	},
 }
+
+func getStopSignal(ctx context.Context, container containerd.Container, defaultSignal syscall.Signal) (syscall.Signal, error) {
```

Comment:
Should we add this to the containerd package and export it?  It seems like people would have to re-implement this each time.

Reply:
Yes, though moving this unmodified into the `containerd` package will create an import cycle; this code depends on `commands.ParseSignal` and `commands` imports `containerd`.  Would you want to move `commands.ParseSignal` (and the associated `commands.signalMap` into `containerd` or should I just return the signal as a `string` and let the caller deal with it?

Reply:
`StopSignal` in the `v1.ImageConfig` is a `string`, so I'd still need to convert that string to a `syscall.Signal` somewhere, likely using `commands.ParseSignal`.  It ends up being the same import cycle problem, but in storing the signal instead of retrieving it.

Reply:
Ok, i see.  I think it would be safe to move the signal map into the containerd client package.  It's generally useful.